### ISSUE TITLE
no-underscore-dangle should not enforce for prototype methods if enforceInMethodNames is false

### ISSUE THREAD

**Issue by harrim91** [eslint/eslint#10755]
**What rule do you want to change?**
`no-underscore-dangle`

**Does this change cause the rule to produce more or fewer warnings?**
Fewer

**How will the change be implemented? (New option, new default behavior, etc.)?**
Update default behaviour of existing option

**Please provide some example code that this change will affect:**
With the method defined in a class no error is raised:
```js
/* eslint no-underscore-dangle: ["error", { "enforceInMethodNames": false }] */
class Foo {
  _bar() {}
}
```
With the prototype defined literally no error is raised:
```js
/* eslint no-underscore-dangle: ["error", { "enforceInMethodNames": false }] */
function Foo() {}

Foo.prototype = {
  _bar () {},
};
```

With the method added to the prototype, an error is raised:
```js
/* eslint no-underscore-dangle: ["error", { "enforceInMethodNames": false }] */
function Foo() {}

Foo.prototype._bar = function () {}
```

**What does the rule currently do for this code?**
If `enforceInMethodNames` option is `false` (the default behaviour), then the first two examples will not raise an error, but the final example will raise an error because the `_bar` method has a dangling underscore.

**What will the rule do after it's changed?**
The examples are equivalent, so the final example should not raise an error.
